pypit/arcomb.py

```diff
@@ -38,84 +39,82 @@ def comb_frames(frames_arr, det, method='weightmean', spect=None, weight=None, f
     if num_frames == 1:
         msgs.info("Only one frame to combine!")
         msgs.info("Returning input frame")
-        return frames_arr[:,:,0]
+        return frames_arr[:, :, 0]
     else:
         msgs.info("Combining {0:d} {1:s} frames".format(num_frames, frametype))
     # Check if the user has allowed the combination of long and short frames (e.g. different exposure times)
     msgs.work("lscomb feature has not been included here yet...")
-    # Check that the type of weights to use is given if the uses wishes to apply a weighted average
-#	if method == 'weightmean' and weights == None:
-#		msgs.error("You must specify the weights if you want a weighted mean combination")
     # Check the user hasn't requested to reject more frames than available
-    if rej_lowhigh[0] > 0 and rej_lowhigh[1] > 0 and rej_lowhigh[0]+rej_lowhigh[1] >= num_frames:
-        msgs.error("You cannot reject more frames than is available with 'rej_lowhigh'."+msgs.newline()+
-                    "There are {0:d} frames and rej_lowhigh will reject {1:d} low and {2:d} high".format(num_frames, rej_lowhigh[0], rej_lowhigh[1]))
+    if reject['lowhigh'][0] > 0 and reject['lowhigh'][1] > 0 and \
+        reject['lowhigh'][0] + reject['lowhigh'][1] >= num_frames:
+        msgs.error("You cannot reject more frames than is available with 'reject lowhigh'." + msgs.newline() +
+                   "There are {0:d} frames and reject lowhigh will reject {1:d} low and {2:d} high".format(num_frames, reject['lowhigh'][0], reject['lowhigh'][1]))
     # Check that some information on the frames was supplied
-    if spect is None:
+    if settings.spect is None:
         msgs.error("When combining the {0:s} frames, spectrograph information".format(frametype)+msgs.newline()+"was not provided")
     # Calculate the values to be used if all frames are rejected in some pixels
-    if set_allrej == 'min':
+    if reject['replace'] == 'min':
         allrej_arr = arcycomb.minmax(frames_arr, 0)
-    elif set_allrej == 'max':
+    elif reject['replace'] == 'max':
         allrej_arr = arcycomb.minmax(frames_arr, 1)
-    elif set_allrej == 'mean':
+    elif reject['replace'] == 'mean':
         allrej_arr = arcycomb.mean(frames_arr)
-    elif set_allrej == 'median':
+    elif reject['replace'] == 'median':
         allrej_arr = arcycomb.median(frames_arr)
-    elif set_allrej == 'weightmean':
+    elif reject['replace'] == 'weightmean':
         msgs.work("No weights are implemented yet")
         allrej_arr = arcycomb.masked_weightmean(frames_arr, maskvalue)
-    elif set_allrej == 'maxnonsat':
-        allrej_arr = arcycomb.maxnonsat(frames_arr, spect['det'][det-1]['saturation']*spect['det'][det-1]['nonlinear'])
+    elif reject['replace'] == 'maxnonsat':
+        allrej_arr = arcycomb.maxnonsat(frames_arr, settings.spect[dnum]['saturation']*settings.spect[dnum]['nonlinear'])
     else:
```

Comment:
These parameters are included in the settings.armlsd file (e.g. `bias combine reject BLAH VALUE`). I've made some changes, as per your previous comment. The code is much cleaner now, so this was a good suggestion :-)